presidio-image-redactor/app.py

```diff
@@ -47,15 +49,23 @@ def redact():
             """Return a redacted image."""
             params = get_json_data(request.form.get("data"))
             color_fill = color_fill_string_to_value(params)
-            image_file = request.files.get("image")
-            if not image_file:
-                raise InvalidParamException("Invalid parameter, please add image data")
-            im = Image.open(image_file)
-
-            redacted_image = self.engine.redact(im, color_fill)
+            if request.get_json(silent=True) and "image" in request.json:
+                im = Image.open(BytesIO(base64.b64decode(request.json.get("image"))))
+                analyzer_entities = request.json.get('analyzer_entities')
+                redacted_image = self.engine.redact(im, color_fill,
+                                                    entities=analyzer_entities)
+                img_byte_arr = image_to_byte_array(redacted_image, im.format)
+                return Response(base64.b64encode(img_byte_arr),
+                                mimetype="application/octet-stream")
 
-            img_byte_arr = image_to_byte_array(redacted_image, im.format)
-            return Response(img_byte_arr, mimetype="application/octet-stream")
+            elif request.files and "image" in request.files:
+                im = Image.open(request.files.get("image"))
+                redacted_image = self.engine.redact(im, color_fill)
```

Comment:
Same comment as above re: `ocr_kwargs` and `**text_analyzer_kwargs`.

Reply:
each arg needs to be parsed separately by the API, like the entities, so I don't think we can do it globally